os/pm/pm_metrics.c

```diff
@@ -17,117 +17,276 @@
  ****************************************************************************/
 
 #include <tinyara/config.h>
+#include <tinyara/pm/pm.h>
 #include <time.h>
 #include <queue.h>
 #include <debug.h>
-#include "pm_metrics.h"
+#include <errno.h>
 #include "pm.h"
 
-time_t time_diff(time_t time1, time_t time2)
+struct pm_metric_s {
+	uint32_t idle_suspend_counts[CONFIG_PM_NDOMAINS];
+	uint32_t wakeup_src_counts[PM_WAKEUP_SRC_COUNT];
+	uint32_t total_suspend_counts;
+	clock_t suspend_ticks[CONFIG_PM_NDOMAINS];
+	clock_t dtime[CONFIG_PM_NDOMAINS];
+	clock_t sticks[PM_COUNT];
+	clock_t stime;
+	clock_t sleep_time;
+};
+
+typedef struct pm_metric_s pm_metric_t;
+
+pm_metric_t *g_pm_metrics;
+
+static void pm_print_metrics(void)
 {
-	if (time1 > time2) {
-		return time1 - time2;
-	} else {
-		return time2 - time1;
+	double suspend_duration;
+	double suspend_percent;
+	double idle_counts_percent;
+	double total_time;
+	double total_wakeup_src_counts;
+	int index;
+	total_time = 0.0;
+	total_wakeup_src_counts = 0.0;
+	for (index = 0; index < PM_WAKEUP_SRC_COUNT; index++) {
+		total_wakeup_src_counts += (double)g_pm_metrics->wakeup_src_counts[index];
+	}
+	for (index = 0; index < PM_COUNT; index++) {
+		total_time += (double)TICK2MSEC(g_pm_metrics->sticks[index]);
+	}
+	pmdbg("\n\n");
+	pmdbg("          DOMAIN                |      SUSPEND TIME      | IDLE SUSPEND COUNTS \n");
```

Comment:
Let's rename SUSPEND TIME to TOTAL SUSPEND TIME

And IDLE SUSPEND COUNTS to ... blocked board sleep count?

Reply:
Won't It opposite?

Board Sleep Count -> no. of times board was able to sleep
Idle Suspend Count -> no. of times board was idle (not sleep) cause of suspended domain